GitUI/CommandsDialogs/SettingsDialog/Pages/ScriptsSettingsPage.cs

```diff
@@ -274,8 +293,8 @@ private void BindScripts(IList<ScriptInfoProxy> scripts, ScriptInfoProxy? select
 
         private void btnAdd_Click(object sender, EventArgs e)
         {
-            ScriptInfoProxy script = _scripts.AddNew();
-            script.HotkeyCommandIdentifier = Math.Max(GitUI.ScriptsEngine.ScriptsManager.MinimumUserScriptID, _scripts.Max(s => s.HotkeyCommandIdentifier)) + 1;
+            ScriptInfoProxy script = _scripts.AddNew()!;
```

Comment:
Why "!"?

Reply:
Nullable annotations on the method say that it can return null. I have no clue when that could happen but am pretty sure that this does not apply to our case so `!` explicitly communicates an expectation that we are sure it won't be null.

Reply:
I know this syntax...

SourceLink looks like non-null:
```
        public T AddNew() => (T)((this as IBindingList).AddNew())!;
```
as well as
```
       private readonly BindingList<ScriptInfoProxy> _scripts = [];
```

Reply:
Hmm, this turned out to be a false-positive. Method information in rider is shown as annotated as possibly returning null while if one goes to the decompiled code, there are no such annotations and the code does not return null at all. So I removed this `!` here.

Reply:
![image](https://github.com/gitextensions/gitextensions/assets/483659/f57f289e-ae27-4268-b258-d1a25cbbf6f5)

¯\\_(ツ)_/¯